tests/testthat/test-get_precipitation_data.R

```diff
@@ -0,0 +1,53 @@
+testthat::test_that("errors", {
+
+  country = "AGO"
+  admin_lvl = "1"
+  startDate = "2022-01-01"
+  endDate = "2022-02-01"
+  id_col = "NAME_2"
+
+  con <- egh_connection(auto_connect = TRUE, use_env = FALSE, user = "epigraph", password = "epigraph")
+  x <- sf::st_read(con,
+                    geometry_column = "geometry",
+                    query = paste0("SELECT * FROM public.", "\"AFG_2\""),
+                    crs = 4326) %>%
+    sf::st_as_sf()
+
+  testthat::expect_error(
```

Comment:
All these tests are checking for proper calling of the function. If a function is not called with the proper arguments, it should raise an Exception. Unit tests can test for that, obviously, but you should have tests that, call the function properly, and check that it does its job accurately. This can be challenging if you have very long function that behave differently depending on how it is called, i.e. are polymorphic. That's why I suggested to break down the `get_precipitation_data` into multiple components, it will require less tests to guarantee that any codepath will behave correctly.